robohive/tutorials/ee_teleop.py

```diff
@@ -251,12 +252,25 @@ def main(env_name, env_args, reset_noise, action_noise, input_device, output, ho
             if ik_result.success==False:
                 print(f"IK(t:{i_step}):: Status:{ik_result.success}, total steps:{ik_result.steps}, err_norm:{ik_result.err_norm}")
             else:
-                act[:7] = ik_result.qpos[:7]
-                act[7:] = gripper_state
-                if action_noise:
-                    act = act + env.env.np_random.uniform(high=action_noise, low=-action_noise, size=len(act)).astype(act.dtype)
-                if env.normalize_act:
-                    act = env.env.robot.normalize_actions(act)
+                if env.env.robot._act_mode == "pos":
+                    act[:7] = ik_result.qpos[:7]
+                    act[7:] = gripper_state
+                    if action_noise:
+                        act = act + env.env.np_random.uniform(high=action_noise, low=-action_noise, size=len(act)).astype(act.dtype)
+                    if env.normalize_act:
+                        act = env.env.robot.normalize_actions(act)
+                elif env.env.robot._act_mode == "vel":
+                    curr_qpos = env.sim.get_state()['qpos'][:7]
+                    target_qpos = ik_result.qpos[:7]
+                    qvel = (target_qpos - curr_qpos)
+                    act[:7] = qvel
+                    act[7:] = gripper_state
+                    if action_noise:
+                        act = act + env.env.np_random.uniform(high=action_noise, low=-action_noise, size=len(act)).astype(act.dtype)
+                    # if env.normalize_act:
```

Comment:
why is this commented out?

To the best of my understanding, this normalization should be there. Also not that in addition to this normalization, this function will also perform action clipping if its outside the specified velocity range.

Reply:
The actuator velocity ranges are defined [here](https://github.com/vikashplus/robohive/blob/6c14798e48b9904dd8c9151837a1d2468b87bcf3/robohive/envs/multi_task/common/kitchen/franka_kitchen.config#L29) for the kitchen tasks.

Reply:
`qvel` wasn't being calculated correctly. I've fixed that by dividing by `env.dt` and now it works correctly after enabling normalization.